### PR TITLE
Upgrade angular to v4-rc-1 and angular-cli to rc-0

### PR DESCRIPTION
**Pull request by maxime1992**
Size of the main app with angular: 
-  2.X : 246KB
- **4.X : 101KB** :tada:

### CODE REVIEW

File: frontend/.angular-cli.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
   "project": {
-    "version": "1.0.0-beta.32.3",
+    "version": "1.0.0-rc.0",
```
Comment (victornoel) [inline frontend/.angular-cli.json]: no need for the version number in this file anymore with rc0 (yep, I read the changelog this morning :P)

File: frontend/yarn.lock [hunk 2]
```diff
@@ -69,59 +69,70 @@
...
-  version "2.4.7"
-  resolved "https://registry.yarnpkg.com/@angular/core/-/core-2.4.7.tgz#8bed9a2d37e81342bd41d189df88aa3e02393cfb"
+"@angular/core@4.0.0-rc.1", "@angular/core@>=2.3.1 <5.0.0 || >=4.0.0-beta <5.0.0":
+  version "4.0.0-rc.1"
+  resolved "https://registry.yarnpkg.com/@angular/core/-/core-4.0.0-rc.1.tgz#7f87b7696b407476e45d6d3c1880a50d5afbb6e3"
 
-"@angular/flex-layout@2.0.0-beta.5":
-  version "2.0.0-beta.5"
-  resolved "https://registry.yarnpkg.com/@angular/flex-layout/-/flex-layout-2.0.0-beta.5.tgz#fd66e094fe6a73664e36aaa6dab6e17b429293c7"
+"@angular/core@^2.0.0":
```
Comment (victornoel) [inline frontend/yarn.lock]: Remove this block, copy the version (`"@angular/core@^2.0.0"`) and paste it just above next to the other angular/core versions:
```
"@angular/core@4.0.0-rc.1", "@angular/core@>=2.3.1 <5.0.0 || >=4.0.0-beta <5.0.0", "@angular/core@^2.0.0":
  version "4.0.0-rc.1"
  resolved "https://registry.yarnpkg.com/@angular/core/-/core-4.0.0-rc.1.tgz#7f87b7696b407476e45d6d3c1880a50d5afbb6e3"
```

Comment (victornoel) [inline frontend/yarn.lock]: And I think that's all, it seems yarn managed to make it work for the others...

File: frontend/.angular-cli.json (hunk 1 above)
Comment (maxime1992) [inline frontend/.angular-cli.json]: I read it too but I must have skipped that part :D ! Well spotted !

File: frontend/yarn.lock (hunk 2 above)
Comment (maxime1992) [inline frontend/yarn.lock]: I think it should be all good now (I rebased)

**Comment by maxime1992** [comment]
I created a new project with ng4 flag and brought back everything that changed

File: frontend/.angular-cli.json [hunk 3]
```diff
@@ -36,12 +36,13 @@
   },
   "lint": [
     {
-      "files": "src/**/*.ts",
-      "project": "src/tsconfig.json"
+      "project": "src/tsconfig.app.json"
```
Comment (victornoel) [inline frontend/.angular-cli.json]: you sure it's not gonna take `node_modules` with it for the linting if you remove the `files` key?

Comment (maxime1992) [inline frontend/.angular-cli.json]: I gave a try and it's not :)
Probably a new conf from latest cli update

**Comment by victornoel** [comment]
By the way, it would be wise to at least upgrade codelizer to 2.0.1 since it is not meant to support angular 4 before this version...

**Comment by maxime1992** [comment]
@victornoel I saw yesterday a [Tweet](https://twitter.com/mgechev/status/835662936106389505) saying that codelyzer was going beta :) ! I did install it and even tho the problem with async has disappeared before, it won't hurt !

**Pull request merged**